ArrayList/DoubleLinkedList.cs

```diff
@@ -1,10 +1,859 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Text;
 
 namespace List
 {
-    class DoubleLinkedList<T> : IList<T> where T : IComparable<T>
+    public class DoubleLinkedList<T> : IList<T> where T : IComparable<T>
     {
+        private int _length;
+        private DoubleLinkedListNode<T> _root;
+        private DoubleLinkedListNode<T> _tail;
+
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            set
+            {
+                _length = value >= 0 ? value : 0;
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    DoubleLinkedListNode<T> currentNode = GetNodeByIndex(index);
+
+                    return currentNode.Value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
```

Comment:
You will always have exception here when you will try to set value. And that's mean that you don't have tests for this part of code